File: java/TuduManager/web.api.spring.boot/src/test/java/net/seymourpoler/tudumanager/web/api/spring/boot/todo/DeleteTodoControllerShould.java
package net.seymourpoler.tudumanager.web.api.spring.boot.todo;

import net.seymourpoler.tudumanager.domain.ErrorCodes;
import net.seymourpoler.tudumanager.domain.ServiceExecutionResult;
import net.seymourpoler.tudumanager.domain.todo.delete.IDeleteTodoService;
import org.junit.Before;
import org.junit.Test;
import org.springframework.http.HttpStatus;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class DeleteTodoControllerShould {

    private IDeleteTodoService service;
    private DeleteTodoController controller;

    @Before
    public void setUp(){
        service = mock(IDeleteTodoService.class);
        controller = new DeleteTodoController(service);
    }

    @Test
    public void return_not_found_when_todo_is_not_found(){
        var errors = List.of(new net.seymourpoler.tudumanager.domain.Error("id", ErrorCodes.NotFound));
        var errorResult = ServiceExecutionResult.of(errors);
        final Integer someId = 23;
        when(service.delete(someId)).thenReturn(errorResult);

        var response = controller.delete(someId);

        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
    }

    @Test
    public void return_ok_when_todo_is_deleted(){
        var executionResult = ServiceExecutionResult.ok();
        final Integer someId = 23;
        when(service.delete(someId)).thenReturn(executionResult);

        var response = controller.delete(someId);

        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
    }
}

File: dotnet/react.mvc-core.postgres.dapper/TuduManager/test/TuduManayer.web.api.Test/Todo/Delete/DeleteTodoControllerShould.cs
using System.Net;
using Microsoft.AspNetCore.Mvc;
using Moq;
using Shouldly;
using TuduManayer.Domain;
using TuduManayer.Domain.Todo.Delete;
using TuduManayer.web.api.Todo.Delete;
using Xunit;

namespace TuduManayer.web.api.Test.Todo.Delete
{
    public class DeleteTodoControllerShould
    {
        private DeleteTodoController controller;
        private Mock<IDeleteTodoService> service;

        public DeleteTodoControllerShould()
        {
            service = new Mock<IDeleteTodoService>();
            controller = new DeleteTodoController(service.Object);
        }

        [Fact]
        public void return_not_found_when_there_is_an_error()
        {
            const int someTodoId = 4;
            service
                .Setup(x => x.Delete(someTodoId))
                .Returns(ServiceExecutionResult.WithErrors());

            var response = controller.Delete(someTodoId) as NotFoundResult;
            
            response.StatusCode.ShouldBe((int)HttpStatusCode.NotFound);
        }
        
        [Fact]
        public void return_ok()
        {
            const int someTodoId = 4;
            service
                .Setup(x => x.Delete(someTodoId))
                .Returns(ServiceExecutionResult.WithSucess());

            var response = controller.Delete(someTodoId) as OkResult;
            
            response.StatusCode.ShouldBe((int)HttpStatusCode.OK);
        }
    }
}